Feature/translation EN > ES digital-ocean.es.md (#1044)
vapor/docs: docs/deploy/digital-ocean.es.md

# Despliegue en DigitalOcean

> Esta guía te mostrará cómo desplegar una aplicación simple de Vapor Hello, world en un [Droplet](https://www.digitalocean.com/products/droplets/). Para seguir esta guía, debes tener una cuenta de [DigitalOcean](https://www.digitalocean.com) con la facturación configurada.

Reviewer: ```suggestion
Esta guía te mostrará cómo desplegar una aplicación simple "Hola mundo" de Vapor en un [Droplet](https://www.digitalocean.com/products/droplets/). Para seguir esta guía, debes tener una cuenta de [DigitalOcean](https://www.digitalocean.com) con la facturación configurada.
```

#### Uso Básico

```sh
$ swiftly install latest

Fetching the latest stable Swift release...
Installing Swift 5.9.1
Downloaded 488.5 MiB of 488.5 MiB
Extracting toolchain...
Swift 5.9.1 installed successfully!

$ swift --version

Swift version 5.9.1 (swift-5.9.1-RELEASE)
Target: x86_64-unknown-linux-gnu
```

## Instalar Vapor con Vapor Toolbox

Ahora que Swift está instalado, instalemos Vapor con Vapor Toolbox. Necesitarás compilar la toolbox desde el código fuente. Consulta las [versiones](https://github.com/vapor/toolbox/releases) de la toolbox en GitHub para encontrar la última versión. En este ejemplo, estamos usando 18.6.0.

### Clonar y Compilar Vapor

Clona el repositorio de Vapor Toolbox.

```sh
git clone https://github.com/vapor/toolbox.git
```

Consulta la última versión.

```sh
cd toolbox
git checkout 18.6.0
```

Compila Vapor y mueve el binario a tu ruta.

```sh
swift build -c release --disable-sandbox --enable-test-discovery
sudo mv .build/release/vapor /usr/local/bin
```

### Crear un Proyecto Vapor

Utiliza el nuevo comando de proyecto de la toolbox para iniciar un proyecto.

```sh
vapor new HelloWorld -n
```

!!! tip "Consejo"

> 	El indicador `-n` brinda una plantilla básica al responder automáticamente no a todas las preguntas.

Reviewer: ```suggestion
	El indicador `-n` brinda una plantilla básica al responder automáticamente "no" a todas las preguntas.
```

![Vapor Splash](../images/vapor-splash.png)

Una vez que finalice el comando, cambia a la carpeta recién creada:

```sh
cd HelloWorld
``` 

### Abrir Puerto HTTP

Para acceder a Vapor en tu servidor, abre un puerto HTTP.

```sh
sudo ufw allow 8080
```

### Ejecutar

Ahora que Vapor está configurado y tenemos un puerto abierto, ejecutémoslo.

```sh
swift run App serve --hostname 0.0.0.0 --port 8080
```

> Visita la IP de tu servidor a través del navegador o terminal local y deberías ver "¡Funciona!". La dirección IP es `134.122.126.139` en este ejemplo.

Reviewer: ```suggestion
Visita la IP de tu servidor a través del navegador o terminal local y deberías ver "It works!". La dirección IP es `134.122.126.139` en este ejemplo.
```

```
$ curl http://134.122.126.139:8080
It works!
```

De vuelta a tu servidor, deberías ver los registros de la solicitud de prueba.

```
[ NOTICE ] Server starting on http://0.0.0.0:8080
[ INFO ] GET /
```

Usa `CTRL+C` para salir del servidor. Puede tomar un segundo apagarlo.

¡Felicitaciones por hacer que tu aplicación Vapor funcione en un Droplet de DigitalOcean!

## Próximos Pasos

El resto de esta guía indica recursos adicionales para mejorar tu despliegue.

### Supervisor

Supervisor es un sistema de control de procesos que puedes ejecutar y monitorear tu ejecutable de Vapor. Con la configuración de Supervisor, tu aplicación puede iniciarse automáticamente cuando se inicia el servidor y reiniciarse en caso de que falle. Obten más información sobre [Supervisor](../deploy/supervisor.md).

### Nginx

> Nginx es un servidor y proxy HTTP extremadamente rápido, probado en batalla y fácil de configurar. Si bien Vapor admite el servicio directo de solicitudes HTTP, el proxy detrás de Nginx puede proporcionar un mayor rendimiento, seguridad y facilidad de uso. Obten más información sobre [Nginx](../deploy/nginx.md).

Reviewer: ```suggestion
Nginx es un servidor y proxy HTTP extremadamente rápido, confiable y fácil de configurar. Si bien Vapor admite el servicio directo de solicitudes HTTP, el proxy detrás de Nginx puede proporcionar un mayor rendimiento, seguridad y facilidad de uso. Obten más información sobre [Nginx](../deploy/nginx.md).
```
